examples/vision/siamese.py

```diff
@@ -0,0 +1,209 @@
+
+"""
+# Similarity Network with Duplet Loss
+
+Author: [Mehdi](https://github.com/s-mrb)<br>
+Date created: 2021/05/02<br>
+Last modified: 2020/05/03<br>
+Description: Similarity Learning Using `Siamese Network` with `Duplet Loss`
+"""
+
+"""
+## Introduction
+
+The [Siamese Network](https://papers.nips.cc/paper/1993/file/288cc0ff022877bd3df94bc9360b9c5d-Paper.pdf) introduced by Bromley and LeCun to solve signature
+verification as an image matching problem is now widely used for finding the
+level of similarity between two images. This algorithm evolved to have many
+different variants, althought the one we are going to use is Duplet Network.
+A duplet network involves two sister netwroks for finding embeddings/encodings
+and then a distance heuristic to find out how much these embeddings differ from
+each other.
+
+"""
+
+"""
+## Necessary Imports
+"""
+
+import random
+import numpy as np
+import tensorflow as tf
+from tensorflow.keras.models import Model
+from tensorflow.keras.layers.experimental.preprocessing import Rescaling
+from tensorflow.keras.layers import Flatten, Dense, Concatenate, Lambda, Input
+from tensorflow.keras.layers import Conv2D, Activation,AveragePooling2D
+from tensorflow.keras.datasets import mnist
+from tensorflow.keras.optimizers import RMSprop
+from tensorflow.keras import utils
+import matplotlib.pyplot as plt
+from keras import backend as K
+
+"""
+## Load data
+"""
+
+(x_train, y_train), (x_test, y_test) = mnist.load_data()
+
+# We change data type to float
+x_train = x_train.astype('float32')
+x_test = x_test.astype('float32')
+
+"""
+## Make Pairs of images
```

Comment:
Suggestion:

- Here and in other section names (`## Subheadings`), please use sentence case
 
```suggestion
## Create pairs of images
```

Reply:
Sure!